framework/src/eventloop/eventloop_common.c

```diff
@@ -0,0 +1,143 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <tinyara/config.h>
+
+#include <debug.h>
+#include <unistd.h>
+#include <queue.h>
+#include <eventloop/eventloop.h>
+
+#include "eventloop_internal.h"
+
+el_app_handle_t g_el_app_list[CONFIG_MAX_TASKS];
+
+#define MAX_PID_MASK   (CONFIG_MAX_TASKS - 1)
+#define APP_IDX_HASH(pid)  ((pid) & MAX_PID_MASK)
+
+el_app_handle_t *eventloop_get_app_handle(void)
+{
+	return &g_el_app_list[APP_IDX_HASH(getpid())];
+}
+
+el_handle_t *eventloop_pop_job(void)
+{
+	return (el_handle_t *)sq_remfirst(&g_el_app_list[APP_IDX_HASH(getpid())].jobs);
+}
+
+int eventloop_push_job(el_handle_t *handle)
+{
+	if (handle == NULL || handle->pid < 0) {
+		return ERROR;
+	}
+
+	sq_addlast((FAR sq_entry_t *)handle, &g_el_app_list[APP_IDX_HASH(handle->pid)].jobs);
+
+	return OK;
+}
+
+int eventloop_job_empty(el_handle_t *handle, bool *result)
+{
+	if (handle == NULL || handle->pid < 0) {
+		return ERROR;
+	}
+
+	*result = sq_empty(&g_el_app_list[APP_IDX_HASH(handle->pid)].jobs);
+
+	return OK;
+}
+
+void eventloop_increase_app_handle_cnt(int pid)
+{
+	if (pid < 0) {
```

Comment:
it is void function, below is enough
```
if (pic >= 0) {
     g_el_app_list.....
}
```
If you want add some error messages, then you can keep your style.